src/screens/ShoppingCart/ShoppingCart.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import {connect} from 'react-redux'
+import Screen from '../../styledComponents/Screen'
+import Button from '../../styledComponents/Button'
+import { Link } from 'react-router-dom'
+import {removeFromCart} from './actionCreators'
+import productsData from '../../data/products'
+import ShoppingCartList from './ShoppingCartList'
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    removeFromCart: (product) => dispatch(removeFromCart(product))
+  }
+}
+
+const mapStateTopProps = (state) => ({
+  productsInCart:state.shoppingCart.productsInCart
+})
+
+
+const ShoppingCart = (props) => {
+
+  const products= {}
+
+  props.productsInCart.map(product => {
+    products[product] ? products[product]++ : products[product] = 1
+  })
+
+  let sum = 0
+  let productsTable = []
+
+  for(const product in products){
+
+    const name =
+      productsData.find(
+        prod => prod.id === parseInt(product)
+      ).name
+
+    const price =
+      productsData.find(
+        prod => prod.id === parseInt(product)
+      ).price
+
+    sum += price * products[product]
+
+    productsTable.push([name,price,products[product],parseInt(product)])
+  }
+
```

Comment:
hmm maybe you can us here some lodash, ramda js helpers 
( IMHO hard to read the code ) 

something around ?
```js

const {productsInCart, sum} = _.reduce(
  _.groupBy(props.productsInCart, 'id'), 
  (acc, products, id) => {
    const product = _.head(products);
    const count = _.size(product);
    return {
      sum: acc.sum + (count * product.price),
      productsInCart: [...acc.productsInCart, {...product, count}]
    }
  }, 
  {productsInCart: []: sum: 0}
);
```